mne/beamformer/_dics.py

```diff
@@ -1,113 +1,440 @@
 """Dynamic Imaging of Coherent Sources (DICS)."""
 
-# Authors: Roman Goj <roman.goj@gmail.com>
+# Authors: Marijn van Vliet <w.m.vanvliet@gmail.com>
+#          Britta Westner
+#          Susanna Aro <susanna.aro@aalto.fi>
+#          Roman Goj <roman.goj@gmail.com>
 #
 # License: BSD (3-clause)
 
 from copy import deepcopy
+import numbers
 
 import numpy as np
-from scipy import linalg
 
 from ..utils import logger, verbose, warn
 from ..forward import _subject_from_forward
 from ..minimum_norm.inverse import combine_xyz, _check_reference
 from ..source_estimate import _make_stc
-from ..time_frequency import CrossSpectralDensity, csd_epochs
-from ._lcmv import _prepare_beamformer_input, _setup_picks, _reg_pinv
+from ..time_frequency import csd_epochs
+from ._lcmv import (_prepare_beamformer_input, _setup_picks, _reg_pinv,
+                    _check_proj_match, _pick_channels_spatial_filter)
 from ..externals import six
 
 
 @verbose
-def _apply_dics(data, info, tmin, forward, noise_csd, data_csd, reg,
-                label=None, picks=None, pick_ori=None, real_filter=False,
-                verbose=None):
-    """Dynamic Imaging of Coherent Sources (DICS)."""
-    is_free_ori, _, proj, vertno, G =\
-        _prepare_beamformer_input(info, forward, label, picks, pick_ori)
+def make_dics(info, forward, csd, reg=0.05, label=None, pick_ori=None,
+              real_filter=False, verbose=None):
+    """Compute Dynamic Imaging of Coherent Sources (DICS) spatial filter.
+
+    Parameters
+    ----------
+    info : dict
+        Measurement info, e.g. ``epochs.info``.
+    forward : dict
+        Forward operator.
+    csd : instance of CrossSpectralDensity
+        The data cross-spectral density (CSD) matrices. A source estimate is
+        performed for each frequency or frequency-bin defined in the CSD
+        object.
+    reg : float
+        The regularization for the cross-spectral density.
+    label : Label | None
+        Restricts the solution to a given label.
+    pick_ori : None | 'normal' | 'max-power'
+        The source orientation to compute the filter for:
+
+            ``None`` : orientations are pooled (Default)
+            'normal' : filters are computed for the orientation tangential to
+                       the cortical surface
+            'max-power' : filters are computer for the orientation that
+                          maximizes spectral power.
```

Comment:
I suspect this will not render so nicely, better to format like the `filters` below